<launch>
  <arg name="start_nodelet_manager" default="true"/>
  <arg name="nodelet_manager_name" default="nodelet_manager"/>
  <arg name="mass"/>

  <node pkg="nodelet"
    type="nodelet"
    name="$(arg nodelet_manager_name)"
    args="manager"
    output="screen"
    if="$(arg start_nodelet_manager)"/>

  <node pkg="nodelet"
    type="nodelet"
    name="trackers_manager"
    args="load trackers_manager/TrackersManager $(arg nodelet_manager_name)"
    clear_params="true"
    output="screen">
    <rosparam file="$(find trackers_manager)/config/trackers.yaml"/>
    <rosparam file="$(find trackers_manager)/config/tracker_params.yaml"/>
    <rosparam file="$(find so3_control)/config/gains.yaml"/>
    <remap from="~odom" to="odom"/>
    <remap from="~cmd" to="position_cmd"/>
  </node>

  <node pkg="nodelet"
    type="nodelet"
    args="load so3_control/SO3ControlNodelet $(arg nodelet_manager_name)"
    name="so3_control"
    required="true"
    clear_params="true"
    output="screen">
    <rosparam file="$(find so3_control)/config/gains.yaml"/>
    <param name="mass" value="$(arg mass)"/>
    <remap from="~odom" to="odom"/>
    <remap from="~position_cmd" to="position_cmd"/>
    <remap from="~so3_cmd" to="so3_cmd"/>
    <remap from="~motors" to="motors"/>
  </node>
</launch>
